File: src/main/kotlin/wowsft/model/gameparams/commander/CrewPersonality.java
package wowsft.model.gameparams.commander;

import wowsft.config.WoWSFT;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class CrewPersonality
{
    private boolean dismissable;
    private boolean hasManyNations;
    private boolean hasOverlay;
    private boolean hasRank;
    @JsonAlias("isPerson")
    private boolean person;
    @JsonAlias("isRetrainable")
    private boolean retrainable;
    @JsonAlias("isUnique")
    private boolean unique;
    private boolean canResetSkillsForFree;
    private String peculiarity;
    private String personName;
    private Ships ships;
    private int costCR;
    private int costELXP;
    private int costGold;
    private int costXP;
}

File: src/main/kotlin/wowsft/model/gameparams/commander/CrewPersonality.kt
package wowsft.model.gameparams.commander

import wowsft.config.WoWSFT
import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class CrewPersonality
{
    var dismissable = false
    var hasManyNations = false
    var hasOverlay = false
    var hasRank = false
    @JsonAlias("person")
    var person = false
    @JsonAlias("retrainable")
    var retrainable = false
    @JsonAlias("unique")
    var unique = false
    var canResetSkillsForFree = false
    var peculiarity: String? = null
    var personName: String? = null
    var ships: Ships? = null
    var costCR = 0
    var costELXP = 0
    var costGold = 0
    var costXP = 0
}
